api/design/type.go

```diff
@@ -4,6 +4,101 @@ import (
 	. "goa.design/goa/v3/dsl"
 )
 
+var Resource = ResultType("application/vnd.hub.resource", func() {
+
+	Description("The resource service provides all resources information.")
+	TypeName("Resource")
+
+	Attribute("id", UInt, "ID is the unique id of the resource", func() {
+		Example("id", 1)
+	})
+
+	Attribute("name", String, "Name of the resource", func() {
+		Example("name", "golang")
+	})
+
+	Attribute("displayName", String, "Display name of the resource", func() {
+		Example("displayName", "golang")
+	})
+
+	Attribute("catalog", Catalog, "Type of catalog where resource belongs")
+
+	Attribute("type", String, "Type of resource", func() {
+		Example("type", "task")
+	})
+
+	Attribute("description", String, "Description of resource", func() {
+		Example("descripiton", "This Task is Golang task to build Go projects")
+	})
+
+	Attribute("latest_version", String, "Latest version o resource", func() {
```

Comment:
`latest_version` can have the object of `version` which will include all info of that particular version - URLs , displayName , description